rust/arrow/src/compute/kernels/aggregate.rs

```diff
@@ -19,9 +19,42 @@
 
 use std::ops::Add;
 
-use crate::array::{Array, PrimitiveArray};
+use crate::array::{Array, LargeStringArray, PrimitiveArray, StringArray};
 use crate::datatypes::ArrowNumericType;
 
+/// Helper macro to perform min/max of strings
+macro_rules! min_max_string_helper {
+    ($array:expr, $cmp:tt) => {{
+        let null_count = $array.null_count();
+
+        if null_count == $array.len() {
+            return None
+        }
+        let mut n = "";
+        let mut has_value = false;
+        let data = $array.data();
+
+        if null_count == 0 {
```

Comment:
so the idea here is that the code is faster if we don't have to check each element for nulls?

Reply:
yes. Generally, operations on non-null fields are faster because there isn't an unpredictable branch on the loop. This check removes that unpredictable branch altogether when there are no nulls.

We could probably still get some more juice by using some of the vertical operations supported by packed_simd.